unittests/CMakeLists.txt

```diff
@@ -7,10 +7,15 @@ if (NOT TARGET gtest)
   include(GoogleTest)
 endif()
 
-set(gtest_libs gtest gtest_main)
-# Clang prior than clang13 (I think) merges both gmock into gtest.
-if (TARGET gmock)
-  list(APPEND gtest_libs gmock gmock_main)
+if(EMSCRIPTEN)
+  set(gtest_libs gtest gmock)
```

Comment:
Hi, 

can someone please explain why we only need `gtest` and `gmock` when it comes to emscripten (and possibly gtest_main and gmock_main too for other platforms) ? 

Sorry I might be missing something because of my limited experience with googletest as a whole.

Reply:
Ahhh is this the reasoning behind this 

https://github.com/compiler-research/CppInterOp/pull/483/files#r1948150573

If yes, I think this is correct. I see why an emscripten build would need a main.cpp !

But again what happens if we link against gtest_main ? We eventually need to call main() isn't it ? 
The flag which ensures main() runs even when using Emscripten’s default JS glue code is `-s INVOKE_RUN=1` so if simply linking with gtest_main doesn't run the test and we need an extra file, we can try building it with the above flag. 

Maybe something for the future if we really don't want the extra file.